addons/web/static/src/legacy/js/core/py_utils.js

```diff
@@ -260,12 +260,12 @@ function pycontext() {
         String(d.getMonth() + 1).padStart(2, "0")}-${
         String(d.getDate()).padStart(2, "0")}`;
     const now = `${
-        String(d.getUTCFullYear()).padStart(4, "0")}-${
-        String(d.getUTCMonth() + 1).padStart(2, "0")}-${
-        String(d.getUTCDate()).padStart(2, "0")} ${
-        String(d.getUTCHours()).padStart(2, "0")}:${
-        String(d.getUTCMinutes()).padStart(2, "0")}:${
-        String(d.getUTCSeconds()).padStart(2, "0")}`;
+        String(d.getFullYear()).padStart(4, "0")}-${
+        String(d.getMonth() + 1).padStart(2, "0")}-${
+        String(d.getDate()).padStart(2, "0")} ${
+        String(d.getHours()).padStart(2, "0")}:${
+        String(d.getMinutes()).padStart(2, "0")}:${
+        String(d.getSeconds()).padStart(2, "0")}`;
```

Comment:
I'm not sure about these. 15.0 is a long term stable and if I'm not mistaken, this code is widely used there, whereas the new py utils (the ones you initially changed) aren't really used in 15.0 (they were introduced there, but really started being used in 16.0). So as it's kind of a tricky change, I wouldn't risk this part. I'm not sure to understand why you had to change them if you don't change legacy tests.

Reply:
I understand that this code is widely used in 15.0 but so was the initial diff in 16.0 (That’s in 16.0 that the bug from the description appears).

If you prefer I can edit only the test file `py_date_tests.js` and it would work with runbot. However, as you mentioned, the tests work on runbot because it is set in UTC. 

If you try running the tests with the system clock at midnight in UTC+2 (for instance) the tests will fail if we don’t change the legacy methods. (That was true before my PR, we just never noticed it). So it would maybe still need a little digging.

Reply:
@Dj0ulo Not if you don't change legacy tests right? My point is, it has always worked like this, and people didn't complain, so I wouldn't change v15 (so, legacy code). I would only change the new implementation.

Reply:
@aab-odoo I removed the modifications I made in the legacy code. (But just to be clear, without them the legacy tests do fail in any case with or without my PR if your system is not in UTC)

Reply:
Ok I'll take a look at that. Thanks